core/arch/arm/include/kernel/pseudo_ta.h

```diff
@@ -19,7 +19,8 @@
 
 #define PTA_ALLOWED_FLAGS	(PTA_MANDATORY_FLAGS | \
 				 TA_FLAG_SECURE_DATA_PATH | \
-				 TA_FLAG_CONCURRENT)
+				 TA_FLAG_CONCURRENT | \
+				 TA_FLAG_KERNEL_DEVICE)
```

Comment:
`_KERNEL_DEVICE` is not very obvious, Kernel here relates to OP-TEE core or REE kernel?
Since this is about discoverablility of the service, why not a flag `TA_FLAG_DISCOVERABLE_SERVICE`?

Reply:
How about `TA_FLAG_DEVICE_ENUM`?